### PR TITLE
Introduce find_forgotten_procs.py

### PR DESCRIPTION
**Pull request by pkoch**
Sometimes Tron loses state and spins up new processes, even though the previous processes are still there and running.

I've been using this script to find those forgotten processes.

### CODE REVIEW

File: contrib/find_forgotten_procs.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+                if inst['node']['hostname'] == hostname
+            ],
+        )
+        for service in services
+    ]
+
+
+def ssh_get_instances(host, service_to_target):
+    remote_output = _check_output(
+        ['ssh', host, 'bash'],
```
Comment (yyejun) [inline contrib/find_forgotten_procs.py]: Add an option to ssh as user@host

File: contrib/find_forgotten_procs.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    signal=DEFAULT_SIGNAL,
+    kill_prefix=DEFAULT_KILL_PREFIX,
+):
+    my_targets = sum((pids for _, pids in results[host]), [])
+
+    if not my_targets:
+        return
+
+    return _check_output(
+        ['ssh', '-A', host, 'bash'],
```
Comment (yyejun) [inline contrib/find_forgotten_procs.py]: Same as above need to ssh as user@host, and -A should be optional

File: contrib/find_forgotten_procs.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    if 'stdout' in kwargs:
+        raise ValueError('stdout argument not allowed, it will be overridden.')
+
+    if 'stdin' in kwargs:
+        curr = kwargs['stdin']
+        if isinstance(curr, basestring):
+            f = TemporaryFile()
+            f.write(curr)
+            f.seek(0)
+            kwargs['stdin'] = f
```
Comment (yyejun) [inline contrib/find_forgotten_procs.py]: Can use StringIO instead?

File: contrib/find_forgotten_procs.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+        for node in service['node_pool']['nodes']
+    ])
+    if target_service_names:
+        target_services = [
+            service
+            for service in services
+            if any(fnmatch(service['name'], t) for t in target_service_names)
+        ]
+    else:
+        target_services = services
```
Comment (yyejun) [inline contrib/find_forgotten_procs.py]: It shouldn't default to all services. Add an option --all instead.

File: contrib/find_forgotten_procs.py (hunk 3 above)
Comment (pkoch) [inline contrib/find_forgotten_procs.py]: Unfortunately, no. It requires an fd.

**Comment by pkoch** [comment]
Added the three requested options.

**Pull request merged**